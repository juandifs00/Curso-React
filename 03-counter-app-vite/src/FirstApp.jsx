import PropTypes from 'prop-types';

const getResult = (a, b) => {
  return (a*b)/(a+b);
}

export const FirstApp = ( {title, subTitle, name} ) => {

  //console.log(props);

  return (
    <>
      {/*<h1>Hello, I'm Juandifs</h1>*/}
      <h1>{ title }</h1>
      {/*<h1>Resultado: {getResult(15,45)}</h1>*/}
      {/*<code> { JSON.stringify(newMessaje) } </code>*/}
      <p>Soy algo que salío aquí</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}
  

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

FirstApp.defaultProps = {
  name: 'Juan Diego RG',
  subTitle: 'No hay un subtítulo',
  //title: 'No hay un título',
}