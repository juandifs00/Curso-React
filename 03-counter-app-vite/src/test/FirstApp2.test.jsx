import { render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Pruebas en FirstApp', () => { 

    const title = 'Hola, soy Juandi';
    const subTitle = 'No hay un subtÃ­tulo';
    
    test('Debe hacer match con el snapshot', () => { 
        
        const { container } = render( <FirstApp title={ title } /> )

        expect ( container ).toMatchSnapshot();

     })
     
     test('Debe de mostrar el mensaje "Hola, soy Juandi"', () => { 

        render( <FirstApp title={ title } /> )

        expect ( screen.getByText( title ) ).toBeTruthy();
        
        // screen.debug();

      })

      test('Debe de mostar el titulo en un h1', () => { 
        
        render( <FirstApp title={ title } /> )

        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );

       })

       test('Debe de mostar el subtitulo enviado por props', () => { 
        
        render( 
            <FirstApp 
                title={ title } 
                subtitle={ subTitle }
            />  
        )

        expect( screen.getAllByText( subTitle ).length ).toBe(4);
        
        })

 })