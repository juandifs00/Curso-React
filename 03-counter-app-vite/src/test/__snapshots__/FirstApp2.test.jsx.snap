// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Juandi
     
  </h1>
  <p>
    Soy algo que salío aquí
  </p>
  <p>
    No hay un subtítulo
  </p>
  <p>
    No hay un subtítulo
  </p>
  <p>
    No hay un subtítulo
  </p>
  <p>
    No hay un subtítulo
  </p>
  <p>
    Juan Diego RG
  </p>
</div>
`;
